### ISSUE TITLE
[HOLD for payment 2023-10-12] [$500] Web - Deleting a manual request in the IOU room that still has a scan request redirects the user to the main chat

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#27862]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open any chat where you can request money
2. Click on the + in the chat and request money
3. Choose manual > Enter the amount > Click Next > the Request
3. Clcik on the + and request money again
4. Choose Scan > Choose file > Then Request
5. Open the IOU report room
6. Delete the manual request

## Expected Result:
User shouldn't be redirected to the main chat because there is still a scan request in the IOU report room

## Actual Result:
 User has been redirected to the main chat

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.72.6
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

[Screencast from 17-09-23 01_19_52.webm](https://github.com/Expensify/App/assets/93399543/80413c87-6dcd-4060-a485-cfd8524dbdb8)

https://github.com/Expensify/App/assets/93399543/58c66e65-06e3-4ae5-aa75-945b29da80ee

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Ahmed-Abdella
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1694903945687269

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~016c0429bfbb307315</li>
        <li>Upwork Job ID: 1704484358540517376</li>
        <li>Last Price Increase: 2023-09-20</li>
<li>Automatic offers: </li>
<ul>
<li>tienifr | Contributor | 26880437</li>
<li>Ahmed-Abdella | Reporter | 26880439</li>
</ul></li>
    </ul>
</details>

**Comment by Ahmed-Abdella** [comment]
Please make sure that only the scan request is left in the IOU after deleting the manual request. To ensure this, make sure that the IOU room has just two requests (one manual and one scan) and delete the manual request.

**Comment by mananjadhav** [comment]
@tienifr's proposal and RCA is correct. Looks good to me.

🎀 👀 🎀 C+ reviewed.

**Comment by youssef-lr** [comment]
LGTM!

**Comment by mananjadhav** [comment]
I was on an unexpected OOO. I reviewed the code change, will work on the checklist today.

**Comment by mananjadhav** [comment]
I don't know at what point should we have added the check for isWhisperAction. @youssef-lr Are you aware? Because otherwise I can't pinpoint the exact PR. We added isWhisper refactor in this [commit](https://github.com/tienifr/App/commit/f6b1eb0a651f2ab0c0cdd3eecf6619c941b9d59b) but that isn't the PR when we should've been adding the Delete check.


Also I don't think we need a regression test here. The delete action was working fine, just not redirecting.
